src/CosmosDB/CosmosDB/SQL/SetAzCosmosDBSqlContainer.cs

```diff
@@ -0,0 +1,174 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+using System.Linq;
+using Microsoft.Azure.Commands.CosmosDB.Models;
+using System.Reflection;
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.CosmosDB.Helpers;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.CosmosDB.Models;
+
+namespace Microsoft.Azure.Commands.CosmosDB
+{
+    [Cmdlet(VerbsCommon.Set, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CosmosDBSqlContainer" , DefaultParameterSetName = NameParameterSet, SupportsShouldProcess = true), OutputType(typeof(PSSqlDatabaseGetResults))]
+    public class SetAzCosmosDBSqlContainer : AzureCosmosDBCmdletBase
+    {
+
+        [Parameter(Mandatory = true, ParameterSetName = NameParameterSet, HelpMessage = Constants.ResourceGroupNameHelpMessage)]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NameParameterSet, HelpMessage = Constants.AccountNameHelpMessage)]
+        public string AccountName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NameParameterSet, HelpMessage = Constants.DatabaseNameHelpMessage)]
+        public string DatabaseName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = Constants.ContainerNameHelpMessage)]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.IndexingPolicyHelpMessage)]
+        public PSSqlIndexingPolicy IndexingPolicy { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.PartitionKeyVersionHelpMessage)]
+        public int? PartitionKeyVersion { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = Constants.PartitionKeyKindHelpMessage)]
+        public string PartitionKeyKind { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = Constants.PartitionKeyPathHelpMessage)]
+        public string[] PartitionKeyPath { get; set; }
```

Comment:
add ValidateNotNullOrEmpty if List<string> Paths cannot be empty.

Reply:
This is handled at our backend. User should not enter "" as partitionkey path. If they do so, they will get an exception thrown from the backend. The exception message is same across various RP platforms, adding a different check here, would remove this consistency.

Reply:
> This is handled at our backend. User should not enter "" as partitionkey path. If they do so, they will get an exception thrown from the backend. The exception message is same across various RP platforms, adding a different check here, would remove this consistency.

ValidateNotNullOrEmpty is a attribute for the property. Refer to https://github.com/Azure/azure-powershell/blob/master/src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs. Search for "ValidateNotNullOrEmpty"

Reply:
Got it. Added it wherever applicable